src/io.rs

```diff
@@ -165,9 +175,9 @@ impl<'l, F: AsFd> Drop for Async<'l, F> {
     fn drop(&mut self) {
         self.inner.kill(&self.dispatcher);
         // restore flags
-        let _ = fcntl_setfl(
+        let _ = set_nonblocking(
             unsafe { BorrowedFd::borrow_raw(self.dispatcher.borrow().fd) },
-            self.old_flags,
+            false,
```

Comment:
This is a slight change in behavior. Currently, if the the FD was already in non-block mode before being wrapped into an `Async<_>`, this is preserved on drop. While with this change, the FD is unconditionally set to blocking mode.

Reply:
The thing with Windows is that it's impossible to tell whether or not a socket is currently in nonblocking mode. The socket can be set to blocking or non-blocking mode, but there is no way to tell what it currently is (without trying to perform a read of course).

So this behavior can't really be ported to Windows. I think the best course of action is to keep the current behavior on Unix, but on Windows to only ever set it to blocking mode.